lib/transport.rb

```diff
@@ -15,6 +17,43 @@ def initialize(max_weight, speed, available)
     @max_weight = max_weight
     @speed = speed.to_f
     @available = available
+    @location = 'in garage'
+    @number_of_deliveries = 0
+    @delivery_cost = 0
+  end
+
+  def self.filter_by(attribute, &block)
+    objects_with_given_attributes = @@park.select { |p| p.respond_to?(attribute) }
+    return objects_with_given_attributes.to_enum(:each) unless block_given?
+
+    result = []
+    objects_with_given_attributes.each do |object|
+      if block.call(object.send(attribute))
+        result << object
+      end
+    end
+    result
+  end
```

Comment:
you can do something like this with  'map', to simplify expression 
objects_with_given_attributes.map {|t| t if block.call(object.public_send(attribute))}

Reply:
done